tools/prowgen/pkg/decorator/requirement.go

```diff
@@ -66,13 +67,13 @@ func ApplyRequirements(job *config.JobBase, requirements, excludedRequirements [
 func resolveRequirements(annotations, labels map[string]string, spec *v1.PodSpec, requirements []spec.RequirementPreset) {
 	if spec != nil {
 		for _, req := range requirements {
-			mergeRequirement(annotations, labels, spec.Containers, &spec.Volumes, req)
+			mergeRequirement(annotations, labels, spec, spec.Containers, &spec.Volumes, req)
 		}
 	}
 }
 
-// mergeRequirement will overlay the requirement on the existing job spec.
-func mergeRequirement(annotations, labels map[string]string, containers []v1.Container, volumes *[]v1.Volume,
+// mergeRequirement will overlay the requirement on the existing job spec. Use mergo for all keys except containers and metadata
+func mergeRequirement(annotations, labels map[string]string, spec *v1.PodSpec, containers []v1.Container, volumes *[]v1.Volume,
 	req spec.RequirementPreset) {
```

Comment:
Just pass `spec` and get containers from that?
```suggestion
func mergeRequirement(annotations, labels map[string]string, spec *v1.PodSpec, volumes *[]v1.Volume,
	req spec.RequirementPreset) {
	containers := spec.Containers
```

Reply:
That won't be helpful. The containers are processed by another function `createContainer` in `generate.go`. I would need to do a rewrite and modify the istio prow configuration.